tools/isobuild/resolver.ts

```diff
@@ -21,18 +21,25 @@ import {
   optimisticReadJsonOrNull,
 } from "../fs/optimistic";
 
-const nativeModulesMap: Record<string, string> = Object.create(null);
-const nativeNames = Object.keys((process as any).binding("natives"));
-
-// Node 0.10 does not include process as a built-in module, but later
-// versions of Node do, and we provide a stub for it on the client.
-nativeNames.push("process");
+// Note: the source code for the following function was copied from the 'builtin-modules' npm package (ver. 3.1.0)
+function getBuiltinModules() {
+  'use strict';
+  const {builtinModules} = require('module');
+
+  const blacklist = [
+    'sys'
+  ];
+
+  // eslint-disable-next-line node/no-deprecated-api
+  return (builtinModules || Object.keys((process as any).binding('natives')))
+      .filter((x: string) => !/^_|^(internal|v8|node-inspect)\/|\//.test(x) && !blacklist.includes(x))
```

Comment:
If you leave out the process.bind fallback the regex test is no longer necessary. And actually the blacklist also doesn't really matter. The sys module exists, but gives a deprecation warning, since node 1.0.0. The NPM sys module is just the util module.

Reply:
OK. The code has been changed to reflect this simplification.